app/logic/Contacts/OWA/OWAAddressbook.ts

```diff
@@ -71,37 +78,71 @@ export class OWAAddressbook extends Addressbook {
       }
       query.Body.IndexedPageItemView.Offset += kMaxCount;
     } while (response.ResultSet.length == kMaxCount);
-    for (let person of persons) {
-      let request: any = {
-        __type: "GetPersonaJsonRequest:#Exchange",
-        Header: {
-          __type: "JsonRequestHeaders:#Exchange",
-          RequestServerVersion: "Exchange2013",
-        },
-        Body: {
-          __type: "GetPersonaRequest:#Exchange",
-          PersonaId: {
-            __type: "ItemId:#Exchange",
-            Id: person.PersonaId.Id,
+    for (let person of this.persons.contents.filter(person => !persons.some(result => result.PersonaId.Id == person.personaID))) {
+      this.persons.remove(person);
+      await SQLPerson.deleteIt(person);
+    }
+    for (let result of persons) {
```

Comment:
In the interest of keeping the functions short and understandable, and hopefully even reusable, could you move the parsing of a person and a group into 2 respective functions, please? Essentially, the part inside the try/catches within the for loops.

Reply:
For consistency with EWS I actually moved the entire loops into their own functions.

Reply:
Makes sense